app/workers/backend_delete_application_worker.rb

```diff
@@ -0,0 +1,38 @@
+# frozen_string_literal: true
+
+class BackendDeleteApplicationWorker < ApplicationJob
+  queue_as :low
+
+  def perform(event_id)
+    event = EventStore::Repository.find_event!(event_id)
+
+    @application = build_application(event)
+
+    [ApplicationKey, ReferrerFilter].each(&method(:delete_backend_associations))
+
+    application.delete_backend_application
+  rescue ActiveRecord::RecordNotFound => exception
+    System::ErrorReporting.report_error(exception, parameters: {event_id: event_id})
+  end
+
+  private
+
+  attr_reader :application
+
+  def build_application(event)
+    app_attrs = {id: event.application.id, application_id: event.application_id}
```

Comment:
I guess this is failure proof because `application` is a `MissingModel::MissingApplication`, right? Not an actual `Cinstance` and therefore there's no risk of the application having been deleted already, correct?

Reply:
Exactly 😄